File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if(s.length() <= 1){
            return s;
        }
        string result = "";
        
        for(int i = 1; i < s.length(); i++){
            int low = i;
            int high = i;
            while(low >= 0 && high < s.length() && s[low] == s[high]){
                string p = s.substr(low, high-low+1);
                low--;
                high++;

                if(p.length() > result.length()){
                    result = p;
                }
            }

            low = i-1;
            high = i;
            while(low >= 0 && high < s.length() && s[low] == s[high]){
                string p = s.substr(low, high-low+1);
                low--;
                high++;
                
                if(p.length() > result.length()){
                    result = p;
                }
            }
        }
        return result;
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def __init__ (self):
        self.ans = ""
    def expand(self, s:str, left: int, right: int) -> None:
        while left >= 0 and right < len(s):
            if s[left] != s[right]:
                break
            left -= 1
            right += 1

        if len(self.ans) < (right - left):
            self.ans = s[left+1:right]
    def longestPalindrome(self, s: str) -> str:
        for i in range(len(s)):
            self.expand(s, i, i)
            self.expand(s,i, i+1)

        return self.ans